messaging/src/main/java/nl/nn/adapterframework/extensions/kafka/KafkaListener.java

```diff
@@ -0,0 +1,205 @@
+
+	/** The group id of the consumer */
+	private @Setter String groupId;
+
+	/** Whether to start reading from the beginning of the topic. */
+	private @Setter boolean fromBeginning = false;
+
+	/** How often to check for new topics when using Patterns. (in MS) */
+	private @Setter int patternRecheckInterval = 5000;
+
+	/**
+	 * The topics to listen to as comma-separated list. Regular expressions are supported,
+	 * for instance: {@code example.*}.
+	 */
+	private @Setter String topics;
+
+	//setter is for testing purposes only.
+	private @Setter(AccessLevel.PACKAGE) Consumer<String, byte[]> consumer;
+	private Iterator<? extends ConsumerRecord<String, byte[]>> waiting;
+	private final Duration pollDuration = Duration.ofMillis(1);
+	private final Map<TopicPartition, OffsetAndMetadata> offsetAndMetadataMap = new HashMap<>();
+	private @Getter(AccessLevel.PACKAGE) Pattern topicPattern;
+	private final Lock lock = new ReentrantLock();
+
+	@Override
+	public void configure() throws ConfigurationException {
+		super.configure();
+		if (StringUtils.isEmpty(groupId))
+			throw new ConfigurationException("groupId must be specified");
+		if (StringUtils.isEmpty(topics))
+			throw new ConfigurationException("topics must be specified");
+		if (patternRecheckInterval < 10)
+			throw new ConfigurationException("patternRecheckInterval should be at least 10");
+
+		properties.setProperty(ConsumerConfig.GROUP_ID_CONFIG, groupId);
+		properties.setProperty(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, fromBeginning ? "earliest" : "latest");
+		properties.setProperty(ConsumerConfig.METADATA_MAX_AGE_CONFIG, String.valueOf(patternRecheckInterval));
+		properties.setProperty(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "false");
+
+		List<String> topicList = StringUtil.split(topics);
+		for (String topic : topicList) {
+			if (!TOPIC_NAME_PATTERN.test(topic))
+				throw new ConfigurationException("topics contains invalid characters. Only a-zA-Z0-9._-* are allowed. (topic: [" + topic + "])");
+		}
+		String pattern = String.join("|", topicList);
+		if (pattern.isEmpty()) throw new ConfigurationException("topics must contain at least one valid topic");
+		topicPattern = Pattern.compile(pattern);
+	}
+
+	@Override
+	public void open() throws ListenerException {
+		lock.lock();
+		try {
+			consumer = buildConsumer();
+			consumer.subscribe(topicPattern);
+			waiting = consumer.poll(Duration.ofMillis(100)).iterator();
+			if (waiting.hasNext()) return;
+			Double metric = (Double) consumer.metrics().values().stream().filter(item -> item.metricName().name().equals("response-total")).findFirst().orElseThrow(() -> new ListenerException("Failed to get response-total metric.")).metricValue();
+			if (metric.intValue() == 0) throw new ListenerException("Didn't get a response from Kafka while connecting for Listening.");
+		} catch (RuntimeException e) {
+			throw new ListenerException(e);
+		} finally {
+			lock.unlock();
+		}
```

Comment:
Ik snap hier de open ook niet. Ik neem aan dat dit een check moet zijn om te kijken of hij verbinding heeft gemaakt. Dat gebeurd toch niet in een future/andere thread?

Reply:
De consumer is in principe niet thread-safe and hoewel `open()` maar vanuit 1 thread gedaan zal worden kan het ook geen kwaad lijkt me, om hier een lock te gebruiken.